Implement the build of the examples

ccamel opened the pull request:
This closes #3

ccamel committed:
[example] #3 Remove examples from repo

The example are now available in a new repo:
=> github.com/crucibuild/agent-go-examples

jpigree commented on Makefile:
```diff
@@ -5,14 +5,24 @@ FULL_GOPATH=${GOPATH}/src/${FULL_NAME}
+build: dependencies	
 	go build -v ./...
 
+examples: dependencies
```
To me this doesn't work. If you do the following, agent-ping will fail to run.(don't forget to clean before doing this)

```
$ make
$ agent-ping
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x7b8d2f]

goroutine 1 [running]:
main.MustOpenResources(0x8a8dc7, 0x18, 0xc420060058, 0x0, 0x7b89e5)
        /home/jp/workspace/perso/go/src/github.com/crucibuild/sdk-agent-go/example/agent-ping/pingAgent.go:38 +0x2f
main.NewPingAgent(0x0, 0xc4200e3f68)
        /home/jp/workspace/perso/go/src/github.com/crucibuild/sdk-agent-go/example/agent-ping/pingAgent.go:56 +0x5b
main.main()
        /home/jp/workspace/perso/go/src/github.com/crucibuild/sdk-agent-go/example/agent-ping/main.go:19 +0x22
```

ccamel replied:
Mmm, I'm pretty sure you have the agent-ping already installed in your go path. Your problem is related with the issue #3 I raised. The agents are not build when simply doing "make".

In my configuration:
```
$ make
$ ./agent-ping
No such file or directory

$ make examples
$ ./agent-ping
Agent responsible for sending ping commands to the pong agent
...
```

That's why I added a specific target in the Makefile for building the examples.

ccamel replied:
Ok. I see. _make_ install the binaries of the examples. So, agree with you, the generation of the resources must be performed first.

I would prefer to have 2 separate phases, one for building the lib, another one for building the examples, if wanted. But I don't sure it is feasible (easily in any case).

jpigree replied:
You can but you will have to list explicitly the packages you want to build in the go build commands. It will be harder to maintain. However, I don't think this sdk is prone to have many more packages.

Why do you prefer to separate the 2 phases?

jpigree replied:
@ccamel do you want me to take over this task?

ccamel replied:
Gladly. But first, I would like to go through the questions raised by #3. As soon as we are ok, you'll be able to go with the fixes in  this branch.

jpigree reviewed (changes requested):
Your commit do improve readability and I quite like it. However, the examples requires resources generation in order to work. So, perhaps you can create a target "resources" which builds resources and you link it to the target "build". When I test, doing "go build ./..." builds the examples and the sdk so we do not need an extra step to build the examples.

jpigree commented:
I also found a pedantic issue. You left several trailing whitespaces in the makefile. Please configure your IDE to automatically remove them. It is sad that Github diffs do not show them.

jpigree committed:
[Makefile] #3 Remove unused variables in Makefile

jpigree committed:
[Makefile] #3 Added a "check" step in the Makefile and link it to the CI

Added a check command which pass linters and various tests on the code.

jpigree committed:
[Makefile] #3 Remove steps wrapping too trivial commands

Go build commands are quite simple so the Makefile should provide steps
for non trivial commands (checks and CI).

This closes #3

jpigree commented:
@ccamel Can you review this a last time? I think we should be good now.

ccamel reviewed (commented):
It's fine.
